lib/datadog/lambda.rb

```diff
@@ -93,22 +94,41 @@ def self.metric(name, value, time: nil, **tags)
     # Generate tags for enhanced metrics
     # @param context [Object] https://docs.aws.amazon.com/lambda/latest/dg/ruby-context.html
     # @return [hash] a hash of the enhanced metrics tags
+    # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
     def self.gen_enhanced_tags(context)
       arn_parts = context.invoked_function_arn.split(':')
-      {
+      # Check if we have an alias or version
+      function_alias = arn_parts[7].nil? ? nil : arn_parts[7]
+
+      tags = {
         functionname: context.function_name,
         region: arn_parts[3],
         account_id: arn_parts[4],
         memorysize: context.memory_limit_in_mb,
         cold_start: @is_cold_start,
-        runtime: "Ruby #{RUBY_VERSION}"
+        runtime: "Ruby #{RUBY_VERSION}",
+        resource: context.function_name
       }
+      # If we have an alias...
+      unless function_alias.nil?
+        # If the alis version is $Latest, drop the $ for ddog tag convention.
+        if function_alias.start_with?('$')
+          function_alias[0] = ''
+          # If the alias is not a version number add the executed version tag
+        elsif function_alias.to_i.zero?
```

Comment:
I don't think this will behave quite as expected if someone has a semantic version as a function alias e.g. '2.3.01', `to_i` will truncate everything after (and including) the first decimal point. 

```
irb(main):010:0> '2.3.01-alpha'.to_i
=> 2
```

Reply:
So it is my understanding that it is not possible. Versions will always be an int and an alias must always start with a letter. The only time this is not the case is when `$Latest` is applied by AWS https://a.cl.ly/2Nu5yqgW.